interfaces/dotnet/Cantera/src/SpeciesCollection.cs

```diff
@@ -64,41 +57,30 @@ public double[] MassFractions
     /// </summary>
     public double[] MoleFractions
     {
-        get => InteropUtil.GetDoubles(_handle, _species.Count,
-            LibCantera.thermo_getMoleFractions);
-
-        set
-        {
-            var retval = LibCantera.thermo_setMoleFractions(_handle,
-                value.Length, value);
+        get => LibCantera.thermo_getMoleFractions(_handle, _species.Count);
```

Comment:
See `MassFractions`.

Reply:
Ok – I have added functionality to automatically get the correct size via an interop call. Howerver, I have changed this class back to the overload that send in a span of known size – for now.